resources/tests/test_resources_models.py

```diff
@@ -73,22 +75,54 @@ def test_berth_is_available_last_season_renew_automatically_invalid_status(
     assert Berth.objects.get(id=berth.id).is_available
 
 
-@freeze_time("2020-01-01T08:00:00Z")
-@pytest.mark.parametrize("status", ["drafted", "offered", "paid", "expired", "refused"])
+@pytest.mark.parametrize("status", ["drafted", "offered", "paid", "error"])
 @pytest.mark.parametrize("renew_automatically", [True, False])
-def test_berth_is_available_ends_during_season(
+def test_berth_is_not_available_ends_during_season_before_lease_ends(
     superuser_api_client, berth, status, renew_automatically
 ):
-    end_date = calculate_berth_lease_end_date()
-    end_date = end_date.replace(month=end_date.month - 1)
+
+    start_date = calculate_berth_lease_start_date()
+    end_date = calculate_berth_lease_end_date() - relativedelta(month=6, day=29)
 
     BerthLeaseFactory(
         berth=berth,
+        start_date=start_date,
         end_date=end_date,
         renew_automatically=renew_automatically,
         status=LeaseStatus(status),
     )
-    assert Berth.objects.get(id=berth.id).is_available
+    with mock.patch(
+        "leases.utils.calculate_berth_lease_start_date"
+    ) as mock_start, mock.patch(
+        "leases.utils.calculate_berth_lease_end_date"
+    ) as mock_end:
+        mock_start.return_value = date(2020, 6, 10)
+        mock_end.return_value = date(2020, 9, 15)
+        assert not Berth.objects.get(id=berth.id).is_available
+
+
+@pytest.mark.parametrize("status", ["drafted", "offered", "paid", "error"])
+@pytest.mark.parametrize("renew_automatically", [True, False])
+def test_berth_is_available_ends_during_season_after_lease_ends(
+    superuser_api_client, berth, status, renew_automatically
+):
+    start_date = calculate_berth_lease_start_date()
+    end_date = calculate_berth_lease_end_date() - relativedelta(month=6, day=29)
+
+    BerthLeaseFactory(
+        berth=berth,
+        start_date=start_date,
+        end_date=end_date,
+        renew_automatically=renew_automatically,
+        status=LeaseStatus(status),
+    )
+    with mock.patch(
+        "leases.utils.calculate_berth_lease_start_date"
+    ) as mock_start, mock.patch(
+        "leases.utils.calculate_berth_lease_end_date"
+    ) as mock_end:
+        mock_start.return_value = date(2020, 6, 30)
+        mock_end.return_value = date(2020, 9, 15)
```

Comment:
Is this test missing the actual testing part :thinking:

Reply:
:d0rp:

Reply:
Fixed ✨